topology_test.go

```diff
@@ -0,0 +1,158 @@
+			fmt.Printf("  |- %s {analysis: %v}\n\n", cc.Name, cc.Summary)
+			for _, ccc := range cc.Components {
+				fmt.Printf("    |- %s {analysis: %v}\n\n", ccc.Name, ccc.Summary)
+				for _, cccc := range ccc.Components {
+					fmt.Printf("      |- %s {analysis: %v}\n\n", cccc.Name, cccc.Summary)
+				}
+			}
+		}
+	}
+}
+
+var _ = ginkgo.Describe("Topology behavior", ginkgo.Ordered, func() {
+
+	ginkgo.It("Should create root tree", func() {
+		tree, err := QueryTopology(TopologyOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		treeJSON, err := json.Marshal(tree)
+		Expect(err).ToNot(HaveOccurred())
+
+		expected := readTestFile("fixtures/expectations/topology_root_tree.json")
+		Expect(expected).Should(MatchJSON(string(treeJSON)))
+	})
+
+	ginkgo.It("Should create child tree", func() {
+		tree, err := QueryTopology(TopologyOptions{ID: dummy.NodeA.ID.String()})
+		Expect(err).ToNot(HaveOccurred())
+
+		treeJSON, err := json.Marshal(tree)
+		Expect(err).ToNot(HaveOccurred())
+
+		expected := readTestFile("fixtures/expectations/topology_child_tree.json")
+		Expect(expected).Should(MatchJSON(string(treeJSON)))
+	})
+
+	ginkgo.It("Should test depth 1 root tree", func() {
+		tree, err := QueryTopology(TopologyOptions{Depth: 1})
+		Expect(err).ToNot(HaveOccurred())
+
+		treeJSON, err := json.Marshal(tree)
+		Expect(err).ToNot(HaveOccurred())
+
+		expected := readTestFile("fixtures/expectations/topology_depth_1_root_tree.json")
+		Expect(expected).Should(MatchJSON(string(treeJSON)))
+	})
+
+	ginkgo.It("Should test depth 2 root tree", func() {
+		tree, err := QueryTopology(TopologyOptions{Depth: 2})
+		Expect(err).ToNot(HaveOccurred())
+
+		treeJSON, err := json.Marshal(tree)
+		Expect(err).ToNot(HaveOccurred())
+
+		expected := readTestFile("fixtures/expectations/topology_depth_2_root_tree.json")
+		Expect(expected).Should(MatchJSON(string(treeJSON)))
+	})
+
+	ginkgo.It("Should test depth 1 tree child tree", func() {
+		tree, err := QueryTopology(TopologyOptions{ID: dummy.LogisticsAPI.ID.String(), Depth: 1})
+		Expect(err).ToNot(HaveOccurred())
+
+		treeJSON, err := json.Marshal(tree)
+		Expect(err).ToNot(HaveOccurred())
+
+		expected := readTestFile("fixtures/expectations/topology_depth_1_child_tree.json")
+		Expect(expected).Should(MatchJSON(string(treeJSON)))
+	})
+
+	ginkgo.It("Should test depth 2 tree child tree", func() {
+		// TODO: Current query with a component_id defined does not return the children if
+		// they are linked via parent_id of that component
+		ginkgo.Skip("SQL Query needs to be fixed for this to work")
+		tree, err := QueryTopology(TopologyOptions{ID: dummy.LogisticsAPI.ID.String(), Depth: 2})
+		Expect(err).ToNot(HaveOccurred())
+
+		treeJSON, err := json.Marshal(tree)
+		Expect(err).ToNot(HaveOccurred())
+
+		expected := readTestFile("fixtures/expectations/topology_depth_2_child_tree.json")
+		Expect(expected).Should(MatchJSON(string(treeJSON)))
+	})
+
+	ginkgo.It("Should test tree with labels", func() {
```

Comment:
I thought we wanted to test json explicitly since the UI will receive these responses as is, but I see your point that having multiple jsons would be difficult.

Will keep this in mind for future tests, and will write a func where just IDs in a tree format can be used to test whether the result is correct/incorrect instead of the entire json